File: 2021/day01/part1.js
const fs = require("fs").promises;

async function main() {
  const input = await fs.readFile("input.txt", "utf8");
  const inputToArray = input.split("\n").filter((val) => val);

  let numOfIncreases = 0;

  for (let i = 0; i < inputToArray.length; i++) {
    const measurement = parseInt(inputToArray[i]);
    const prevMeasurement = parseInt(inputToArray[i - 1]);

    if (prevMeasurement) {
      if (measurement > prevMeasurement) {
        numOfIncreases = numOfIncreases + 1;
      }
    }
  }

  console.log(numOfIncreases);
}

main();

File: 2023/day04/part1.py
def read_file():
  with open('input.txt', encoding="utf-8") as f:
    return f.read()
  
def main():
  input = read_file()
  cards = input.split("\n")
  total_points = 0

  for card in cards:
    if card:
      card_numbers = card.split(": ")[1].split(" | ")
      winning_numbers = list(filter(bool, card_numbers[0].split(" ")))
      chosen_numbers = list(filter(bool, card_numbers[1].split(" ")))

      points = 0
      for winning_number in winning_numbers:
        if winning_number in chosen_numbers:
          if points == 0: points = 1
          else: points = points * 2

          print('match!', winning_number, points)

      total_points += points

  print(total_points)

if __name__ == '__main__':
  main()